pymodelextractor/teachers/pac_comparison_strategy.py

```diff
@@ -0,0 +1,51 @@
+from pythautomata.utilities.uniform_length_sequence_generator import UniformLengthSequenceGenerator
+from pythautomata.base_types.alphabet import Alphabet
+from pythautomata.base_types.sequence import Sequence
+from math import ceil, log, comb
+import numpy as np
+
+
+class PACComparisonStrategy:
+    def __init__(self, target_model_alphabet: Alphabet, epsilon: float, delta: float, max_seq_length: int = 128, 
+                compute_epsilon_star: bool = True, sequence_generator = None):
+        self._epsilon = epsilon
+        self._delta = delta
+        self._equivalence_queries_count = 0
+        self._compute_epsilon_star = compute_epsilon_star
+
+        if sequence_generator is None:
+            self._sequence_generator = UniformLengthSequenceGenerator(target_model_alphabet, max_seq_length)
+        else:
+            self._sequence_generator = sequence_generator
+
+    def get_counterexample_between(self, model, target_model) -> Sequence:
```

Comment:
types of `model` and `target_model`

Reply:
This method doesn't have types so that we can use PAC comparison strategy with every state machine type we have defined. Perhaps we could define an abstraction that contains every type of state machine.

Reply:
can't we at the very least add a type to model?